### PR TITLE
IR-5050 migrate helm setting to engine setting 

### PR DESCRIPTION
**Pull request by mshafiqmk**
## References
closes #_insert number here_

### CODE REVIEW

**Comment by barankyle** [comment]
This needs to be updated off of dev to resolve merge conflicts/

File: packages/common/src/schemas/setting/helm-setting.schema.ts [hunk 1]
```diff
@@ -29,10 +29,6 @@ import { getValidator, querySyntax, Type } from '@feathersjs/typebox'
 
 import { dataValidator, queryValidator } from '../validators'
 
-export const helmSettingPath = 'helm-setting'
-export const helmMainVersionPath = 'helm-main-version'
-export const helmBuilderVersionPath = 'helm-builder-version'
-
 export const helmSettingMethods = ['find', 'get', 'create', 'patch', 'remove'] as const
```
Comment (hanzlamateen) [inline packages/common/src/schemas/setting/helm-setting.schema.ts]: Why this whole schema file is not removed?

File: packages/server-core/src/projects/project/project-helper.ts [hunk 2]
```diff
@@ -186,9 +197,9 @@ export const updateBuilder = async (
           `kubectl delete deployment --ignore-not-found=true ${builderDeployments.body.items[0].metadata!.name}`
         )
 
-      if (helmSettings && helmSettings.builder && helmSettings.builder.length > 0)
+      if (helmSettings.length > 0 && helmBuilder)
```
Comment (hanzlamateen) [inline packages/server-core/src/projects/project/project-helper.ts]:
```suggestion
      if (helmSettings.length > 0 && helmBuilder && helmBuilder.length > 0)
```

File: packages/server-core/src/setting/helm-setting/helm-setting.ts [hunk 3]
```diff
@@ -39,38 +27,16 @@ Infinite Reality Engine. All Rights Reserved.
...
 
 declare module '@ir-engine/common/declarations' {
   interface ServiceTypes {
-    [helmSettingPath]: HelmSettingService
     [helmMainVersionPath]: { find: () => Promise<string[]> }
     [helmBuilderVersionPath]: { find: () => Promise<string[]> }
   }
 }
 
 export default (app: Application): void => {
```
Comment (hanzlamateen) [inline packages/server-core/src/setting/helm-setting/helm-setting.ts]: Why is this file not completely removed?

File: scripts/fetch-helm-versions.ts [hunk 4]
```diff
@@ -62,13 +63,18 @@ cli.main(async () => {
...
+
+    const helmBuilder = helmSettings.find((setting) => setting.key == EngineSettings.Helm.Main)?.value
+    const helmMain = helmSettings.find((setting) => setting.key === EngineSettings.Helm.Builder)?.value
 
     const helmMainVersionName = path.join(appRootPath.path, 'helm-main-version.txt')
     const helmBuilderVersionName = path.join(appRootPath.path, 'helm-builder-version.txt')
 
     if (helmSettings) {
-      if (helmSettings.main && helmSettings.main.length > 0) fs.writeFileSync(helmMainVersionName, helmSettings.main)
+      if (helmMain) fs.writeFileSync(helmMainVersionName, helmMain)
```
Comment (hanzlamateen) [inline scripts/fetch-helm-versions.ts:77]: Put a length check here too

File: scripts/fetch-helm-versions.ts [hunk 5]
```diff
@@ -79,8 +85,7 @@ cli.main(async () => {
           }
         }
       }
-      if (helmSettings.builder && helmSettings.builder.length > 0)
-        fs.writeFileSync(helmBuilderVersionName, helmSettings.builder)
+      if (helmBuilder) fs.writeFileSync(helmBuilderVersionName, helmBuilder)
```
Comment (hanzlamateen) [inline scripts/fetch-helm-versions.ts]: Put a length check here too

**Review by hanzlamateen — CHANGES_REQUESTED** [review]

File: packages/server-core/src/setting/helm-setting/helm-setting.ts (hunk 3 above)
Comment (mshafiqmk) [reply packages/server-core/src/setting/helm-setting/helm-setting.ts]: this service is getting helm relation information from a helm repo , we need to keep it.

Comment (hanzlamateen) [reply packages/server-core/src/setting/helm-setting/helm-setting.ts]: For this you should introduce a new service name `helm-version` and in that service you should introduce the code. The endpoint should have take a query parameter `action` which will have string enum value of `main` or `builder` based on which it should return the helm version.

This service will be somewhat similar to how `packages/server-core/src/integrations/metabase/metabase-url` service is written. As it also makes an external call.

File: scripts/fetch-helm-versions.ts (hunk 5 above)
Comment (hanzlamateen) [reply scripts/fetch-helm-versions.ts]: @mshafiqmk I think this is not addressed yet

Comment (mshafiqmk) [reply scripts/fetch-helm-versions.ts]: you mean on this line ?

``` if (helmBuilder) fs.writeFileSync(helmBuilderVersionName, helmBuilder)```

Comment (hanzlamateen) [reply scripts/fetch-helm-versions.ts]: made changes here: https://github.com/ir-engine/ir-engine/pull/570/commits/a5264d89b659fbf55e6afcd5db3f287c92004b09

**Comment by hanzlamateen** [comment]
Converted to draft untill https://github.com/ir-engine/ir-engine/pull/570/files#r1842155736 is addressed.

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**